### PR TITLE
Optional data sea ice thickness for atmosphere-only configuration

### PR DESCRIPTION
**Pull request by sanAkel**
# Objectives:
- Provide a `logical` **option**: `SEAICE_THICKNESS_EXT_DATA` to set sea ice thickness from data, using `ExtData`.
- Relevant to **atmosphere-only** model configuration.
- **Reproduces current answers** with default setting of ⬆️ `SEAICE_THICKNESS_EXT_DATA: .FALSE.`

# How was it accomplished?

- When **non-default** `SEAICE_THICKNESS_EXT_DATA: .TRUE.` is set, sea ice thickness data is expected; it is read via `ExtData.` Therefore it must be [NetCDF](https://www.unidata.ucar.edu/software/netcdf/) formatted; it can not be `binary`, because no support to read using `MAPL_ReadForcing` is provided.

- In order to accomplish ⬆️ objective(s), again using logical `SEAICE_THICKNESS_EXT_DATA`, the _friendliness_ of variables: `HI, TI, SI` has been _protected_ so that they are friendly across [GEOS_SaltWaterGridComp](https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/develop/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOSsaltwater_GridComp) and [GEOSdataseaice_GridComp](https://github.com/GEOS-ESM/GEOSgcm_GridComp/tree/develop/GEOSogcm_GridComp/GEOSseaice_GridComp/GEOSdataseaice_GridComp) only for _default_ value of `SEAICE_THICKNESS_EXT_DATA: .FALSE.`, **which reproduces answers**. 
- If you aren't sure of their present _friendly_ connectivity, see https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/86ede1666e607d3116224caa9519fefd88be8410/GEOS_GcmGridComp.F90#L1971-L1976 One reason for not having this friendliness across sea ice (`GEOSdataseaice_GridComp`) and atmosphere (`GEOS_SaltWaterGridComp`) is to eliminate [no-work-no-change](https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/86ede1666e607d3116224caa9519fefd88be8410/GEOSogcm_GridComp/GEOS_OgcmGridComp.F90#L46-L50). This is now accomplished via selection of the **non-default** option for `SEAICE_THICKNESS_EXT_DATA: .TRUE.`

- In the case of **NON-default**, i.e., `SEAICE_THICKNESS_EXT_DATA: .TRUE.`,
   1. One reads data sea ice thickness (as additional) data in `GEOS_DataSeaIceGridComp_ExtData.F90`. It is exported to [GEOSsaltwater_GridComp](https://github.com/GEOS-ESM/GEOSgcm_GridComp/tree/develop/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOSsaltwater_GridComp) and used in [GEOS_SimpleSeaiceGridComp.F90](https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/develop/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOSsaltwater_GridComp/GEOS_SimpleSeaiceGridComp.F90) to update ice surface temperature (`TSKINI`). Additional exports: sea ice thickness and snow are also added.
   2. **Reason for not adding `SEAICE_THICKNESS_EXT_DATA` logic to existing `GEOS_DataSeaIceGridComp.F90` is to eliminate clutter and provide a clean code for future development**, thus all of it is in `GEOS_DataSeaIceGridComp_ExtData.F90`
   3. The _switch_ between `GEOS_DataSeaIceGridComp_ExtData.F90` and `GEOS_DataSeaIceGridComp.F90` is accomplished via new logic: `DATA_SEAICE` in [CMakeLists.txt](https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/feature/sanAkel/ext_SIT-data/GEOSogcm_GridComp/GEOSseaice_GridComp/GEOSdataseaice_GridComp/CMakeLists.txt). See ⬇️ for instructions.

# How was it tested?

## Default build:
- [**See here for full instructions**](https://github.com/GEOS-ESM/GEOSgcm#run-cmake) `cmake .. -DBASEDIR=$BASEDIR/Linux -DCMAKE_Fortran_COMPILER=ifort -DCMAKE_INSTALL_PREFIX=../install`
  - **Default** `SEAICE_THICKNESS_EXT_DATA: .FALSE.` **For this no change in [AGCM.rc] is needed**.
     ✅ **All answers are bitwise identical to current develop**.
  - **NON-Default** `SEAICE_THICKNESS_EXT_DATA: .TRUE.` set in `AGCM.rc`. As mentioned ⬆️ this is not allowed because only non-default build with`GEOS_DataSeaIceGridComp_ExtData.F90` supports this setting in `AGCM.rc`.
     ✅  **Run time abort [via ASSERT](https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/f39788342d088e5d2c1b3ea39e63d778212317ee/GEOSogcm_GridComp/GEOSseaice_GridComp/GEOSdataseaice_GridComp/GEOS_DataSeaIceGridComp.F90#L119)**. With this we do not need any `#ifdef`.

## Non-default build:
- Add `-DDATA_SEAICE=VERSION_2` to **CMake**: `cmake .. -DBASEDIR=$BASEDIR/Linux -DCMAKE_Fortran_COMPILER=ifort -DCMAKE_INSTALL_PREFIX=../install -DDATA_SEAICE=VERSION_2`.  **Note**: If you are _testing_ this new option (`SEAICE_THICKNESS_EXT_DATA: .TRUE.`), you may want to _adjust_ arguments to `-DCMAKE_INSTALL_PREFIX` so as to have two build and install directories.
   - **Default** `SEAICE_THICKNESS_EXT_DATA: .FALSE.` **For this no change in [AGCM.rc] is needed**.
      ✅ **All answers are bitwise identical to current develop**. This is expected because `SEAICE_THICKNESS_EXT_DATA: .FALSE.` does everything as in current develop ➡️ @lcandre2 an easy way to test and compare! 😄 
   -  ✅ **NON-Default** `SEAICE_THICKNESS_EXT_DATA: .TRUE.` set in `AGCM.rc` yields _new_ answers from the _new_ implementation ➡️ **ready for _science_testing_**.

### CODE REVIEW

File: GEOSogcm_GridComp/GEOSseaice_GridComp/GEOSdataseaice_GridComp/GEOS_DataSeaIceGridComp.F90 [hunk 1]
```diff
@@ -112,6 +113,11 @@ subroutine SetServices ( GC, RC )
     VERIFY_(STATUS)
 
     call MAPL_GetResource ( MAPL,    ocean_extData, Label="OCEAN_EXT_DATA:",   DEFAULT=.FALSE., __RC__ ) ! .TRUE. or .FALSE.
+    call MAPL_GetResource ( MAPL,    seaIceT_extData, Label="SEAICE_THICKNESS_EXT_DATA:",  DEFAULT=.FALSE., _RC ) ! .TRUE. or .FALSE.
+    if (seaIceT_extData) then
+      if(MAPL_AM_I_ROOT()) print *, 'You have set SEAICE_THICKNESS_EXT_DATA: .TRUE. For this feature, a _newer_: version-2 of data sea ice must be used.'
+      _ASSERT(.FALSE.,'Recompile and try.')
```
[the remark this answers is not in the document]
Comment (sanAkel) [reply GEOSogcm_GridComp/GEOSseaice_GridComp/GEOSdataseaice_GridComp/GEOS_DataSeaIceGridComp.F90]: Thank you @mathomp4, @tclune

**Review by yvikhlya — DISMISSED** [review]

**Review by zhaobin74 — APPROVED** [review]

**Review by lcandre2 — DISMISSED** [review]
Approve, but there may be a need for another pull request once I test these changes with my sea ice thickness product.

**Comment by mathomp4** [comment]
Borrowing this PR to fix the GitHub CI

**Review by sdrabenh — APPROVED** [review]

**Review by mathomp4 — APPROVED** [review]

**Comment by sdrabenh** [comment]
Works fine for the default build. I trust @sanAkel for the non-default option

**Pull request merged**